Fix i18n date

nicolaskempf57 opened the pull request:
Replace datagouv/components date functions with native ones

ThibaudDauce reviewed (changes requested):
On http://dev.local:3000/fr/beta/admin/site/users/ I have:

> RangeError: date value is not finite in DateTimeFormat.format()

nicolaskempf57 commented:
> On http://dev.local:3000/fr/beta/admin/site/users/ I have:
> 
> > RangeError: date value is not finite in DateTimeFormat.format()

I added a `null` case to the function to avoid the error. And I fixed the users page using the wrong attribute
